### PR TITLE
Fix the not locked when there are explicit files.

### LINKED ISSUE #100453: Kubectl .lock file is ambiguous
#### What happened:

``` bash
KUBECONFIG=/dir1/config kubectl config use-context ctx --kubeconfig=/dir2/config
```
Use `/dir1/config.lock` and `/dir2/config`

#### What you expected to happen:

Use `/dir2/config.lock` and `/dir2/config`

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #100453

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/clientcmd/config.go [hunk 1]
```diff
@@ -169,15 +169,23 @@ func NewDefaultPathOptions() *PathOptions {
 // modified element.
 func ModifyConfig(configAccess ConfigAccess, newConfig clientcmdapi.Config, relativizePaths bool) error {
 	if UseModifyConfigLock {
-		possibleSources := configAccess.GetLoadingPrecedence()
-		// sort the possible kubeconfig files so we always "lock" in the same order
-		// to avoid deadlock (note: this can fail w/ symlinks, but... come on).
-		sort.Strings(possibleSources)
-		for _, filename := range possibleSources {
+		if configAccess.IsExplicitFile() {
```
Comment (BenTheElder) [inline staging/src/k8s.io/client-go/tools/clientcmd/config.go:172]: Is this check really sufficient to ensure that none of the other files will be read / written to below?
I don't think it is.

Why should possible sources contain files we won't actually read|write if that is the case? This seems brittle.

Comment (wzshiming) [reply staging/src/k8s.io/client-go/tools/clientcmd/config.go:172]: Only `writeCurrentContext` and `writePreferences` use `GetLoadingPrecedence `, and `IsExplicitFile` is also used first

https://github.com/kubernetes/kubernetes/blob/1822523afb68f33df242b1c43590154f1a361784/staging/src/k8s.io/client-go/tools/clientcmd/config.go#L196-L206


https://github.com/kubernetes/kubernetes/blob/1822523afb68f33df242b1c43590154f1a361784/staging/src/k8s.io/client-go/tools/clientcmd/config.go#L382-L407
https://github.com/kubernetes/kubernetes/blob/1822523afb68f33df242b1c43590154f1a361784/staging/src/k8s.io/client-go/tools/clientcmd/config.go#L444-L465

**Comment by wzshiming** [comment]
I found it has been fixed in 1.20, https://github.com/kubernetes/kubernetes/pull/93293
Close this

**Pull request closed without merging**